openlibrary/plugins/admin/code.py

```diff
@@ -829,6 +829,48 @@ def GET(self):
                 return f.read()
 
 
+class pd_dashboard:
+    def GET(self):
+        def sort_requests(_pd_requests):
+            results = {
+                "requested_access": [],
+                "emailed": [],
+                "fulfilled": [],
+            }
+            for r in _pd_requests:
+                status = r.get("notifications", {}).get("rpd")
+                if status == 0:
+                    results["requested_access"].append(r)
+                elif status == 1:
+                    results["emailed"].append(r)
+                elif status == 2:
+                    results["fulfilled"].append(r)
+            return results
+
+        def make_preference_query(_offset: int) -> list:
+            return web.ctx.site.things(
+                {
+                    "type": "/type/object",
+                    "key~": "/people/*/preferences",
+                    "notifications": {"pda~": "*"},
+                    "limit": max_limit,
+                    "offset": _offset,
```

Comment:
```suggestion
                    "offset": _offset,
                    "created>": "2025-05-01T00:00:00.000000",
```
Test restricting query to recently created objects.